Assets/Packages/com.studio23.ss2.authsystem.steam/Runtime/Data/SteamAuthProvider.cs

```diff
@@ -0,0 +1,106 @@
+using System;
+using Cysharp.Threading.Tasks;
+using Steamworks;
+using Steamworks.Data;
+using Studio23.SS2.AuthSystem.Data;
+using UnityEngine;
+
+namespace Studio23.SS2.AuthSystem.Steam.Data
+{
+    [CreateAssetMenu(fileName = "Steam Auth Provider", menuName = "Studio-23/AuthSystem/Providers/Steam", order = 1)]
+    public class SteamAuthProvider : ProviderBase
+    {
+
+        public uint AppId;
+
+        void OnEnable()
+        {
+            _providerType = ProviderTypes.Steam;
```

Comment:
This should not be set OnEnable. This should be an abstract property in providerbase